test/FlashLoanV1Router02.spec.ts

```diff
@@ -0,0 +1,317 @@
+          token.address,
+          tokenAmount,
+          wallet.address,
+          MaxUint256,
+          overrides
+        )
+      )
+        .to.emit(token, 'Transfer')
+        .withArgs(wallet.address, pool.address, tokenAmount)
+        .to.emit(pool, 'Transfer')
+        .withArgs(AddressZero, AddressZero, MINIMUM_LIQUIDITY)
+        .to.emit(pool, 'Transfer')
+        .withArgs(AddressZero, wallet.address, expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+        .to.emit(pool, 'Sync')
+        .withArgs(tokenAmount)
+        .to.emit(pool, 'Mint')
+        .withArgs(router.address, tokenAmount)
+
+      expect(await pool.balanceOf(wallet.address)).to.eq(expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+    })
+
+    it('addLiquidityETH', async () => {
+      const ETHAmount = expandTo18Decimals(1)
+
+      const expectedLiquidity = expandTo18Decimals(1)
+      await expect(
+        router.addLiquidityETH(
+          wallet.address,
+          MaxUint256,
+          { ...overrides, value: ETHAmount }
+        )
+      )
+        .to.emit(WETHPool, 'Transfer')
+        .withArgs(AddressZero, AddressZero, MINIMUM_LIQUIDITY)
+        .to.emit(WETHPool, 'Transfer')
+        .withArgs(AddressZero, wallet.address, expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+        .to.emit(WETHPool, 'Sync')
+        .withArgs(ETHAmount)
+        .to.emit(WETHPool, 'Mint')
+        .withArgs(
+          router.address,
+          ETHAmount
+        )
+
+      expect(await WETHPool.balanceOf(wallet.address)).to.eq(expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+    })
+
+    async function addLiquidity(tokenAmount: BigNumber) {
+      await token.transfer(pool.address, tokenAmount)
+      await pool.mint(wallet.address, overrides)
+    }
+    it('removeLiquidity', async () => {
+      const tokenAmount = expandTo18Decimals(1)
+      await addLiquidity(tokenAmount)
+
+      const expectedLiquidity = expandTo18Decimals(1)
+      await pool.approve(router.address, MaxUint256)
+      await expect(
+        router.removeLiquidity(
+          token.address,
+          expectedLiquidity.sub(MINIMUM_LIQUIDITY),
+          wallet.address,
+          MaxUint256,
+          overrides
+        )
+      )
+        .to.emit(pool, 'Transfer')
+        .withArgs(wallet.address, pool.address, expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+        .to.emit(pool, 'Transfer')
+        .withArgs(pool.address, AddressZero, expectedLiquidity.sub(MINIMUM_LIQUIDITY))
+        .to.emit(token, 'Transfer')
+        .withArgs(pool.address, wallet.address, tokenAmount.sub(1000))
+        .to.emit(pool, 'Sync')
+        .withArgs(1000)
+        .to.emit(pool, 'Burn')
+        .withArgs(router.address, tokenAmount.sub(1000), wallet.address)
+
+      expect(await pool.balanceOf(wallet.address)).to.eq(0)
+      const totalSupplyToken = await token.totalSupply()
+      expect(await token.balanceOf(wallet.address)).to.eq(totalSupplyToken.sub(1000))
+    })
+
+    it('removeLiquidityETH', async () => {
```

Comment:
This is not changed, why test again in 02?

Reply:
Deleted.